src/ring_theory/dedekind_domain/ideal.lean

```diff
@@ -730,3 +731,42 @@ begin
 end
 
 end is_dedekind_domain
+
+section height_one_spectrum
+
+/-!
+### Height one spectrum of a Dedekind domain
+If `R` is a Dedekind domain of Krull dimension 1, the maximal ideals of `R` are exactly its nonzero
+prime ideals.
+We define `height_one_spectrum` and provide lemmas to recover the facts that prime ideals of height
+one are prime and irreducible. -/
+
+namespace is_dedekind_domain
+
+/-- The height one prime spectrum of a Dedekind domain `R` is the type of nonzero prime ideals of
+`R`. Note that this equals the maximal spectrum if `R` has Krull dimension 1. -/
+@[nolint has_inhabited_instance]
+def height_one_spectrum := {v : prime_spectrum R // v.val ≠ 0 }
```

Comment:
It depends how you plan to develop this further, but
```lean
structure height_one_spectrum :=
(to_ideal : ideal R)
(prime : to_ideal.is_prime)
(ne_bot : to_ideal ≠ ⊥)
```
will certainly make what you have so far more readable

Reply:
Thanks for the suggestion. I have slightly changed the proposed names to make them more consistent with `prime_spectrum`.